### PR TITLE
Allow wolfSSL_RSA_public_decrypt on Windows.

### PR DESCRIPTION
**Pull request by haydenroche5**

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -48611,9 +48611,8 @@ int wolfSSL_RSA_private_decrypt(int len, const unsigned char* fr,
     return ret;
 }
 
-#if !defined(_WIN32) && !defined(HAVE_SELFTEST) && \
-    (!defined(HAVE_FIPS) || \
-        (defined(HAVE_FIPS_VERSION) && HAVE_FIPS_VERSION > 2))
+#if !defined(HAVE_SELFTEST) && (!defined(HAVE_FIPS) || \
```
Comment (dgarske) [inline src/ssl.c:48614]: @haydenroche5  Can you also update the comment on the `#endif`? It still has `_WIN32`.

**Review by dgarske — CHANGES_REQUESTED** [review]
Also curious if you looked at the other occurrences of `_WIN32` in this file? They seems to be related to `vsnprintf`. Oddly there is a `XSNPRINTF`, which seems to properly handle the Windows side of things. It might be good to look at those while in there testing.

**Comment by haydenroche5** [comment]
I'll check out the other _WIN32 instances and fix the #endif comment. Thanks David.

File: src/ssl.c [hunk 2]
```diff
@@ -29067,7 +29067,7 @@ int wolfSSL_BIO_vprintf(WOLFSSL_BIO* bio, const char* format, va_list args)
             break;
 
         case WOLFSSL_BIO_MEMORY:
-    #if defined(OPENSSL_EXTRA) && (!defined(_WIN32) || defined(HAVE_VA_COPY))
```
Comment (haydenroche5) [inline src/ssl.c:29070]: Windows has versions of va_copy, vsnprintf, and va_end, so I don't think this needs to be gated on !_WIN32 anymore. The library built fine for me on Windows in Visual Studio. I also built the library in Cygwin and ran the unit tests and those passed.

This was the only instance of HAVE_VA_COPY in the project, so I got rid of it.

**Comment by haydenroche5** [comment]
According to [this SO answer](https://stackoverflow.com/a/25827156), the va_* functions are available starting in Visual Studio 2013. I believe we use Visual Studio 11.0, AKA Visual Studio 2012, in the Jenkins "Visual Studio Build Test" (just looking at an example build log), so that would explain why we're seeing a failure here.

@kaleb-himes do you happen to know if there's any particular need to support this older VS version, or if we could upgrade to a more recent one?

**Comment by kaleb-himes** [comment]
@haydenroche5,

Yes we have customers that use VS 2012 and we retain backwards compatibility for the sake of those users. Some day it will be deprecated but I am not sure when that day will come.

- K

**Review by dgarske — APPROVED** [review]

**Pull request merged**